extension/background/intentParser.js

```diff
@@ -20,34 +20,73 @@ export function registerMatcher(intentName, matcher) {
   }
   INTENT_NAMES.push(intentName);
   INTENT_NAMES.sort();
-  INTENTS[intentName] = { matcher };
+  INTENTS[intentName] = { matcher, followUp };
 }
 
 let initialized = false;
 let phraseSet;
 
 function initialize() {
   const phrases = [];
+  const followUpPhrases = {};
+
   for (const name in INTENTS) {
-    const { matcher } = INTENTS[name];
+    const { matcher, followUp } = INTENTS[name];
+
     for (const line of splitPhraseLines(matcher)) {
       const compiled = compile(line, {
         entities: entityTypes,
         intentName: name,
       });
+
       INTENTS[name].compiledMatcher = compiled;
-      phrases.push(compiled);
+      phrases.push({ compiled });
+    }
+
+    if (followUp) {
+      followUpPhrases[name] = [];
+      // Some intents may need be able to return to the base intent.
+      // For example, `search.searchPage` can be followed up by `search.next`.
+      // A user may then wish to execute another search before the popup closes,
+      // but it will be treated as a follow up, when it should reset to the base intent
+      followUpPhrases[name].push({ compiled: INTENTS[name].compiledMatcher });
+
+      for (const nextPhrase of followUp) {
+        const desc = {
+          followUpLabel: nextPhrase.label,
+          followUpUtterance: true,
+        };
+
+        if (nextPhrase.intent) {
```

Comment:
The initial implementation would allow an intent to set a different intent as follow up. Now that any intent can be called while the mic is open, this will no longer be required.